src/PolyhedralGeometry/helpers.jl

```diff
@@ -6,7 +6,12 @@ function assure_matrix_polymake(m::Union{AbstractMatrix{Any}, AbstractMatrix{Fie
     a, b = size(m)
     if a > 0
         i = findfirst(_cannot_convert_to_fmpq, m)
-        m = Polymake.Matrix{scalar_type_to_polymake[typeof(m[i])]}(m)
+        t = typeof(m[i])
+        if t <: Union{values(scalar_type_to_polymake)...}
+            m = Polymake.Matrix{Polymake.convert_to_pm_type(t)}(m)
```

Comment:
Huh, what's going on here? So we have this definition:
```julia
const scalar_type_to_polymake = Dict{Type, Type}([(fmpq, Polymake.Rational),
                                    (nf_elem, Polymake.QuadraticExtension{Polymake.Rational}),
                                    (Union{fmpq, nf_elem}, Polymake.QuadraticExtension{Polymake.Rational}),    # needed for Halfspace{nf_elem} etc
                                    (Float64, Float64)])
```

And now... if `t` is one of the type in the values, such as `Float64` or `Polymake.Rational`, then we use `Polymake.Matrix{Polymake.convert_to_pm_type(t)}`.

But isn't that call to `Polymake.convert_to_pm_type` redundant for the arguments on the right side?

So, this would also work?
```suggestion
            m = Polymake.Matrix{t}(m)
```

Reply:
This has something to do with julia's type system, the way `CxxWrap` works with that system and `Polymake.jl`'s definitions for constructors (in this case the matrix):
`t` is an explicit type here, which for example could be a `Polymake.QuadraticExtensionAllocated{...}`, an explicit subtype of the abstract type `Polymake.QuadraticExtension{...}`. This kind of type setup always happens when we wrap our types with `CxxWrap`. In the general cases this makes no big difference (like the usual dispatch), but the `Polymake.Matrix{T}` constructor we want to use re-directs the call to a wrapped C++ function, which is only defined for `QuadraticExtension` and not `QuadraticExtensionAllocated` due to the way the types are linked. Thus we take advantage of the `convert_to_pm_type(t)` function of `Polymake.jl` which changes the type to its abstract version. We probably could adjust this behavior in `Polymake.jl` (@benlorenz what do you think?), but with the current release of `Polymake.jl` i assume it is the cleanest option to implement this.

Reply:
I think it would make sense to add some workarounds for that in Polymake.jl, but for now we do need this.

Reply:
OK, I see, thank you for explaining